### DISCUSSION TITLE
Getting "conan install failed" while building milvus from source code(with gcc 7.5)

### DISCUSSION THREAD

**Discussion by IcyXi** [milvus-io/milvus#36079] [Q&A and General discussion]
The instructions on GitHub state that before building from source code, you need to ensure that gcc is version 7.5. However, when I run the make command, I get the following error。It seems I need to install a more higher version of gcc. Could you plesse help? Thanks! 

Error:
arrow/15.0.0: Invalid ID: arrow/15.0.0 requires C++17, which your compiler does not support.
conan install failed

My OS is Ubuntu 22.04

The followings are the requirements for Linux on GitHub:
go: >= 1.21
cmake: >= 3.26.4
gcc: 7.5
python: > 3.8 and  <= 3.11
https://github.com/milvus-io/milvus/blob/master/DEVELOPMENT.md#prerequisites

**Comment by xiaofan-luan** [comment]
gcc 7.5 might not fully support c++17 features, which is conflicted with arrows requirements.
Could you try to upgrade to gcc9 or higher? I'll change the docuemnt

**Reply by UntidyRAM** [reply]
A bit late to the party considering the age of the post, but I had the same issue with `gcc 7.5`.

I used `gcc 13.4` and the build did not get stuck there anymore. This was on openSUSE (with some adjustments to `install_deps.sh` to use `zypper`).

I suggest a change to https://github.com/milvus-io/milvus/blob/master/DEVELOPMENT.md#prerequisites would be good.

Thanks for the help.
